backend/src/main/java/org/cryptomator/hub/api/UsersResource.java

```diff
@@ -79,9 +79,12 @@ public Response putMe(@Nullable @Valid UserDto dto) {
 		user.setPictureUrl(jwt.getClaim("picture"));
 		user.setEmail(jwt.getClaim("email"));
 		if (dto != null) {
-			user.setEcdhPublicKey(dto.ecdhPublicKey);
-			user.setEcdsaPublicKey(dto.ecdsaPublicKey);
-			user.setPrivateKeys(dto.privateKeys);
+			if (user.getEcdhPublicKey() == null && user.getEcdsaPublicKey() == null && user.getPrivateKeys() == null) {
+				user.setEcdhPublicKey(dto.ecdhPublicKey);
+				user.setEcdsaPublicKey(dto.ecdsaPublicKey);
+				user.setPrivateKeys(dto.privateKeys);
+				eventLogger.logUserAccountSetupComplete(jwt.getSubject(), jwt.getName());
+			}
```

Comment:
After upgrading from 1.3.x to 1.4.x, each user will get a new ECDSA key on first login.

This is done via `addEcdsaKeyIfMissing()`:

https://github.com/cryptomator/hub/blob/c18af8c3ee69e2ed0c8fa15cf5cd789282ae25ce/frontend/src/common/userdata.ts#L89-L137

I.e. we rely on the ability to update at least this key, even if the other keys are already set. This would be a rather specific condition though, therefore I like it better to allow changing *any* user property when making a PUT request. Makes a clearer API.

Reply:
If the sole purpose of the newly added `if` is to log setup completion, I guess it requires something like this:

```java
boolean wasUninitialized = user.getPrivateKeys() == null;
// ...
if (wasUninitialized && user.getPrivateKeys() != null) {
  // log
}
```

Reply:
Agree, with https://github.com/cryptomator/hub/blob/581186fd860df20e2597822f2a47585b38296451/backend/src/main/java/org/cryptomator/hub/api/UsersResource.java#L83-L93 we now log the changes but do not prevent it.